lib/spack/spack/cmd/load.py

```diff
@@ -57,10 +57,18 @@ def setup_parser(subparser):
         "--first",
         action="store_true",
         default=False,
-        dest="load_first",
+        dest="load_best",
         help="load the first match if multiple packages match the spec",
     )
 
+    subparser.add_argument(
+        "--best",
```

Comment:
I believe
```suggestion
        "--best", "--first"
```
is supported, then you don't have to repeat yourself